doc/SONiC_OC_NTP_HLD.md

```diff
@@ -0,0 +1,491 @@
+
+With this feature, users will be provided the same capabilities via Management CLI, REST and gNMI using OpenConfig Yang models.
+It also provides users to configure NTP source and NTP vrf (Management VRF for now?) for addtional ntp control.
+
+## 1.1 Requirements
+
+### 1.1.1 Front end configuration and get capabilities
+
+#### 1.1.1.1 add/delete NTP server      
+This installs NTP server information in the Redis ConfigDB and in the ntp.conf file (NTP configuration file).   
+The NTP server can be IPv4 address, IPv6 address , or the name of a NTP server.   
+Mutliple NTP servers can be configured.
+  
+#### 1.1.1.2 add NTP source
+This installs the global NTP source information in the Redis ConfigDB and in the ntp.conf file. It enables ntpd to listen to the specific source.      
+NTP source can be configured as either IP address or interface name.    
+Only one NTP source will be installed as the global NTP source. A new configured NTP source will override the existing NTP source.   
+  
+#### 1.1.1.3 delete NTP source
+This deletes the global NTP source entry from the Redis ConfigDB and from the ntp.conf file.   
+
+#### 1.1.1.4 add/delete VRF name
+This installs the global NTP VRF information in the Redis ConfigDB. It is used by the ntp configuration script to start the ntpd in a specific VRF context.   
+For this release, only Management VRF and default instance are supported(?). When the global NTP VRF is deleted, ntpd will be restarted in the default instance.
+
+#### 1.1.1.5 Get NTP association
+This displays the output of "ntpq -p" command.   
+
+### 1.1.2 Backend mechanisms to support configuration and get
+
+#### 1.1.2.1 add/delete NTP server
+The creates or deletes a NTP server entry in the Redis ConfigDB.
+
+```
+  "NTP_SERVER|10.11.0.1": {
+    "type": "hash",
+    "value": {
+      "NULL": "NULL"
+    }
+  },
+  "NTP_SERVER|2001:aa:aa::a": {
+    "type": "hash",
+    "value": {
+      "NULL": "NULL"
+    }
+  },
+  "NTP_SERVER|pool.ntp.org": {
+    "type": "hash",
+    "value": {
+      "NULL": "NULL"
+    }
+  }
+```
+
+A change in the NTP_SERVER entry triggers hostcfgd to start the NTP configuration script, which in turn writes each NTP server to the ntp.conf and then restart the ntp service.   
+   
+SONiC click CLI only supports IP address for NTP server. It can be extended to server name as well.   
+
+
+#### 1.1.2.2 add/delete NTP source
+
+This creates or deletes a global NTP source entry in the Redis ConfigDB. The NTP source can be a L3 interface name or an interface IP address (IPv4 or IPv6).   
+
+```
+  "NTP|global": {
+    "type": "hash",
+    "value": {
+      "source": "Ethernet36"
+    }
+  }
+```
+
+A change in this entry triggers hostcfgd to start the NTP configuration script, which in turn writes the ntp source to the ntp.conf and then restart the ntp service.   

```

Comment:
Sorry for the late reply. When NTP source intf/ip is configured, ntp.conf will be configured to listen on that interface, besides 127.0.0.1. If mgmt intf is not configured, user can configure eth0 as source intf so that ntp would listen on eth0 instead of Loopback0. Is that working for your case?

Reply:
I don't have the newer code yet, so cannot confirm. But, i take it what appears to be described is the DHCP strategy (i.e. there is no MGMT_INTERFACE, but there might be some non-Loopback0 interfaces) so with the implementation of source interface, the admin would be able to configure that, and that interface (eth0 in this case) would show up as a listen directive in ntp.conf. I take it  that this is the workaround for fixing the ntp.conf.j2 template.

Reply:
The selection log in ntp.conf.j2 will be updated to take the user configured source-interface first, in this case eth0